developer-docs-site/static/move_demo/sources/CAPMessageBoard.move

```diff
@@ -0,0 +1,141 @@
+        message: vector<u8>,
+        board: address
+    }
+
+    /// init the message board and move the resource to signer
+    public fun message_board_init_internal(account: &signer){
+        let board = CapMessageBoard {
+            notice: Vector::empty<u8>()
+        };
+        move_to(account, board);
+        let notice_cap = NoticeChangeCapability{ is_active: true};
+        move_to(account, notice_cap);
+        let notice_cap_container = NoticeChangeCapabilityContainer {addr: Signer::address_of(account)};
+        move_to(account, notice_cap_container);
+        move_to(account, NoticeChangeEventHandle {
+            change_events:  Event::new_event_handle<NoticeChangeEvent>(account)
+        });
+        move_to(account, NoticeCapEventHandle {
+            change_events:  Event::new_event_handle<NoticeCapEvent>(account)
+        })
+
+    }
+
+    /// expose the function as a transaction interface
+    public(script) fun message_board_init(account: signer) {
+        message_board_init_internal(&account);
+    }
+
+    public fun view_message(board_addr: address) : vector<u8> acquires CapMessageBoard {
+        let board = borrow_global<CapMessageBoard>(board_addr);
+        board.notice
+    }
+
+    /// participant can register to
+    public fun register_internal(participant: &signer){
+        move_to(participant, NoticeChangeEventHandle {
+            change_events:  Event::new_event_handle<NoticeChangeEvent>(participant)
+        })
+    }
+
+    public(script) fun register(participant: signer){
+        register_internal(&participant);
+    }
+
+    /// board owner control adding new participants
+    public(script) fun add_participant(account: signer, participant: address) acquires NoticeCapEventHandle,NoticeChangeCapabilityContainer {
+        add_participant_internal(&account, participant);
+    }
+
+    public fun add_participant_internal(account: &signer, participant: address) acquires NoticeCapEventHandle, NoticeChangeCapabilityContainer {
+        let notice_cap_container = borrow_global_mut<NoticeChangeCapabilityContainer>(Signer::address_of(account));
+        notice_cap_container.addr = participant;
+        let event_handle = borrow_global_mut<NoticeCapEventHandle>(Signer::address_of(account));
+
+        Event::emit_event<NoticeCapEvent>(
+            &mut event_handle.change_events,
+            NoticeCapEvent {
+                timestamp: Timestamp::now_seconds(),
+                participant,
+                board: Signer::address_of(account)
+            }
+        );
+    }
+
+    public(script)  fun claim_notice_cap(account: signer, board: address) acquires NoticeChangeCapabilityContainer {
+        claim_notice_cap_internal( &account, board);
+    }
+
+    /// claim notice capability
+    public fun claim_notice_cap_internal(account: &signer, board: address) acquires NoticeChangeCapabilityContainer{
+        let notice_cap_container = borrow_global<NoticeChangeCapabilityContainer>(board);
```

Comment:
this container should be removed once claimed?

Reply:
this account here is the participant.  NoticeChangeCapabilityContainer is owner by board owner. Participant won't be able to delete board_owner's resource right?

Reply:
remove this contain and use offer instead